tests/raftstore/server.rs

```diff
@@ -167,7 +167,21 @@ impl Simulator for ServerCluster {
 
         conn.set_read_timeout(Some(timeout)).unwrap();
         let mut resp_msg = Message::new();
-        let get_msg_id = try!(rpc::decode_msg(&mut conn, &mut resp_msg));
+        let get_msg_id = try!(rpc::decode_msg(&mut conn, &mut resp_msg).map_err(|e| {
+            if let CodecError::Io(ref err) = e {
+                // If read timeout, the error will throw "Resource temporarily unavailable",
```

Comment:
For socket, this message is certainly corresponding to EAGAIN and EWOULDBLOCK for different  system, so we can use error code and message to check this only.